provider/ocm_policies_data_source.go

```diff
@@ -0,0 +1,194 @@
+	logger       logging.Logger
+	awsInquiries *cmv1.AWSInquiriesClient
+}
+
+func (t *OcmPoliciesDataSourceType) GetSchema(ctx context.Context) (result tfsdk.Schema,
+	diags diag.Diagnostics) {
+	result = tfsdk.Schema{
+		Description: "List of rosa operator role for a specific cluster.",
+		Attributes: map[string]tfsdk.Attribute{
+			"operator_role_policies": {
+				Description: "Operator role policies.",
+				Attributes:  operatorRolePoliciesNames(),
+				Computed:    true,
+			},
+			"account_role_policies": {
+				Description: "Account role policies.",
+				Attributes:  accountRolePoliciesNames(),
+				Computed:    true,
+			},
+		},
+	}
+	return
+}
+
+func accountRolePoliciesNames() tfsdk.NestedAttributes {
+	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+		"sts_installer_permission_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"sts_support_permission_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"sts_instance_worker_permission_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"sts_instance_controlplane_permission_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+	})
+}
+
+func operatorRolePoliciesNames() tfsdk.NestedAttributes {
+	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+		"openshift_cloud_credential_operator_cloud_credential_operator_iam_ro_creds_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"openshift_cloud_network_config_controller_cloud_credentials_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"openshift_cluster_csi_drivers_ebs_cloud_credentials_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"openshift_image_registry_installer_cloud_credentials_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"openshift_ingress_operator_cloud_credentials_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+		"openshift_machine_api_aws_cloud_credentials_policy": {
+			Type:     types.StringType,
+			Computed: true,
+		},
+	})
+}
+
+func (t *OcmPoliciesDataSourceType) NewDataSource(ctx context.Context,
+	p tfsdk.Provider) (result tfsdk.DataSource, diags diag.Diagnostics) {
+	// Cast the provider interface to the specific implementation:
+	parent := p.(*Provider)
+
+	// Get the collection of clusters:
+	awsInquiries := parent.connection.ClustersMgmt().V1().AWSInquiries()
+
+	// Create the resource:
+	result = &OcmPoliciesDataSource{
+		logger:       parent.logger,
+		awsInquiries: awsInquiries,
+	}
+	return
+}
+
+func (t *OcmPoliciesDataSource) Read(ctx context.Context, request tfsdk.ReadDataSourceRequest,
+	response *tfsdk.ReadDataSourceResponse) {
+	// Get the state:
+	state := &OcmPoliciesState{}
+	diags := request.Config.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	policiesResponse, err := t.awsInquiries.STSPolicies().List().Send()
```

Comment:
This request get the actual policy document, but there's also
`stsCredentialResponse, err := c.ocm.ClustersMgmt().
		V1().
		AWSInquiries().
		STSCredentialRequests().
		List().
		Parameter("is_hypershift", isHypershift).
		Send()`
		
Which I think could be used to remove the constant for the policy names